online-shop/src/components/ProductDetails/ProductDetails.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { Product, ProductArray } from '../../model/model';
+import './ProductDetails.scss';
+import { Link } from 'react-router-dom';
+import { API_PRODUCTS } from '../../util/API';
+
+export { ProductDetails };
+
+class ProductDetails extends React.Component<{ productId: number, shoppingCart: ProductArray, deleteItem: any }, { product: Product }> {
+   
+    constructor(props: any) {
+        super(props);
+        this.state = ({ 
+            product: new Product(0, "LOADING", "LOADING", 0, "LOADING", "LOADING",)
```

Comment:
It would be nicer to have a `loading` flag and show something like a loading indicator instead of the fields (a simple `if` statement inside the render).